pkg/apis/core/validation/namespacedcloudprofile_test.go

```diff
@@ -0,0 +1,1093 @@
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	gomegatypes "github.com/onsi/gomega/types"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/utils/ptr"
+
+	"github.com/gardener/gardener/pkg/apis/core"
+	. "github.com/gardener/gardener/pkg/apis/core/validation"
+)
+
+var _ = Describe("NamespacedCloudProfile Validation Tests ", func() {
+	Describe("#ValidateNamespacedCloudProfile", func() {
+		It("should forbid empty NamespacedCloudProfile resources", func() {
+			namespacedCloudProfile := &core.NamespacedCloudProfile{
+				ObjectMeta: metav1.ObjectMeta{},
+				Spec:       core.NamespacedCloudProfileSpec{},
+			}
+
+			errorList := ValidateNamespacedCloudProfile(namespacedCloudProfile)
+
+			Expect(errorList).To(ConsistOf(
+				PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":  Equal(field.ErrorTypeRequired),
+					"Field": Equal("metadata.name"),
+				})), PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":  Equal(field.ErrorTypeRequired),
+					"Field": Equal("metadata.namespace"),
+				})), PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":  Equal(field.ErrorTypeNotSupported),
+					"Field": Equal("spec.parent.kind"),
+				})), PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":  Equal(field.ErrorTypeRequired),
+					"Field": Equal("spec.parent.name"),
+				}))))
+		})
+
+		It("should allow NamespacedCloudProfile resource with only name and parent field", func() {
+			namespacedCloudProfile := &core.NamespacedCloudProfile{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "profile",
+					Namespace: "default",
+				},
+				Spec: core.NamespacedCloudProfileSpec{
+					Parent: core.CloudProfileReference{
+						Kind: "CloudProfile",
+						Name: "profile-parent",
+					},
+				},
+			}
+
+			errorList := ValidateNamespacedCloudProfile(namespacedCloudProfile)
+			Expect(errorList).To(BeEmpty())
+
+			newNamespacedCloudProfile := namespacedCloudProfile.DeepCopy()
+			newNamespacedCloudProfile.Spec.Parent.Name = "cloud-profile-parent-1"
+			newNamespacedCloudProfile.ResourceVersion = "2"
+
+			errorList = ValidateNamespacedCloudProfileUpdate(newNamespacedCloudProfile, namespacedCloudProfile)
+			Expect(errorList).To(ConsistOf(
+				PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":  Equal(field.ErrorTypeInvalid),
+					"Field": Equal("spec.parent"),
+				}))))
+		})
+
+		It("should not allow a NamespacedCloudProfile to change its parent reference", func() {
+			namespacedCloudProfile := &core.NamespacedCloudProfile{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "profile",
+					Namespace: "default",
+				},
+				Spec: core.NamespacedCloudProfileSpec{
+					Parent: core.CloudProfileReference{
+						Kind: "CloudProfile",
+						Name: "profile-parent",
+					},
+				},
+			}
+
+			errorList := ValidateNamespacedCloudProfile(namespacedCloudProfile)
+			Expect(errorList).To(BeEmpty())
+		})
```

Comment:
Shouldn't this case be tested with `ValidateNamespacedCloudProfileUpdate` function?

Reply:
Correct, probably forgot to add a TODO there, fixed in https://github.com/gardener/gardener/pull/9440/commits/026d5bdc62ce06a524e6839f6d663a7dcf36a6e6